plugins/grpc/csharp/v1.65.4/Dockerfile

```diff
@@ -17,10 +17,16 @@ WORKDIR /build/grpc
 RUN bazel build //src/compiler:grpc_plugin_support
 RUN bazel build //src/compiler:grpc_csharp_plugin.stripped
 
+FROM mcr.microsoft.com/dotnet/sdk:8.0.401-bookworm-slim@sha256:8c6beed050a602970c3d275756ed3c19065e42ce6ca0809f5a6fcbf5d36fd305 AS dotnetrestore
+WORKDIR /build
+COPY --link ./build.csproj /build/build.csproj
+RUN mkdir /nuget && dotnet restore --packages /nuget
+
 FROM gcr.io/distroless/cc-debian12:latest@sha256:3b75fdd33932d16e53a461277becf57c4f815c6cee5f6bc8f52457c095e004c8 AS base
 
 FROM scratch
 COPY --link --from=base / /
+COPY --link --from=dotnetrestore --chmod=0755 /nuget /nuget
```

Comment:
We should leave the chmod off here - all the files within don't need their permissions changed.

Reply:
Dropped from all affected Dockerfiles in https://github.com/bufbuild/plugins/pull/1429/commits/65378b84d700c5d3f89cf7e579e43c92c41edd6f — I'll update the other PRs as well.